extensions/haskell_toolchains.bzl

```diff
@@ -0,0 +1,165 @@
+""" Module extension to install bindist haskell toolchains"""
+
+load(
+    "@rules_haskell//haskell:ghc_bindist.bzl",
+    "ghc_bindist",
+    "ghc_bindist_toolchain_declaration",
+    "ghc_bindists_toolchain_declarations",
+    "haskell_register_ghc_bindists",
+)
+
+_bindists_tag = tag_class(
+    attrs = {
+        "version": attr.string(
+            doc = "[see rules_haskell_toolchains](toolchain.html#rules_haskell_toolchains-version)",
+        ),
+        "ghcopts": attr.string_list(
+            doc = "[see rules_haskell_toolchains](toolchain.html#rules_haskell_toolchains-ghcopts)",
+        ),
+        "haddock_flags": attr.string_list(
+            doc = "haddock_flags: [see rules_haskell_toolchains](toolchain.html#rules_haskell_toolchains-haddock_flags)",
+        ),
+        "repl_ghci_args": attr.string_list(
+            doc = "[see rules_haskell_toolchains](toolchain.html#rules_haskell_toolchains-repl_ghci_args)",
+        ),
+        "cabalopts": attr.string_list(
+            doc = "[see rules_haskell_toolchains](toolchain.html#rules_haskell_toolchains-cabalopts)",
+        ),
+        "locale": attr.string(
+            doc = "[see rules_haskell_toolchains](toolchain.html#rules_haskell_toolchains-locale)",
+        ),
+    },
+    doc = """See [rules_haskell_toolchains](toolchain.html#rules_haskell_toolchains).
+    Only the first `bindists` tag is taken into account (according to the iteration order over modules).
+    """,
+)
+
+_bindist_tag = tag_class(
+    attrs = {
+        "name": attr.string(
+            mandatory = True,
+            doc = "A unique name for the repository",
```

Comment:
I think this needs a bit more explanation. As I understand it, it's only a unique name for the tag within the calling module. I.e. it doesn't have to be a globally unique name, correct?